test/test.js

```diff
@@ -235,6 +235,37 @@ describe('Page', function() {
     }));
   });
 
+  describe('Browser.Events.disconnected', function() {
+    it('should emitted when browser gets closed', SX(async function() {
```

Comment:
this one fired `disconnected` twice, i notice that. Same effect when i use the `browser`, Assuming it added to two instance the browser and originalBrowser. For me it's important you to know, i am wrong please leave some suggestion.

Reply:
Good catch!

That's because:
1. Connection should reset it's `this._ws` to `null` once the COnnection._onClose method is called.
2. Connection._onClose should remove added event listeners (use `helper.addEventListener` / `helper.removeEventListeners` for this)
2. The Connection.dispose() method should fast-return if this._ws === null.

Reply:
I am not fully understand it, but i do the trick. Hopefully good to go :smile: